### PR TITLE
Turn on navigation check on request form

### PR DESCRIPTION
**Pull request by mkuklis**
I wish all form/navigation related issues were that simple...

### CODE REVIEW

File: src/RequestForm.js [hunk 1]
```diff
@@ -858,7 +858,7 @@ export default stripesForm({
   form: 'requestForm',
   asyncValidate,
   asyncBlurFields: ['item.barcode', 'requester.barcode'],
-  navigationCheck: false,
+  navigationCheck: true,
```
Comment (zburke) [inline src/RequestForm.js]: This one was bugging me and it took me a while to remember why. See the conversation about it under https://github.com/folio-org/ui-requests/pull/204 / https://issues.folio.org/browse/UIU-675. 

Maybe we should leave it alone for now?

Comment (mkuklis) [reply src/RequestForm.js]: Oh no. Your intuition was right @zburke :(. It looks like what was done in #204 is a work around but not a real fix. I will take a quick look at it today. If it's really impossible to fix it we should leave it alone for now.

Comment (mkuklis) [reply src/RequestForm.js]: @zburke I made a small change which will reinitialize the form when the `Block details` button is clicked. That should allow us to leave the request form without seeing the navigation check modal. This is not ideal but I think it's still better than completely turning the check off.

Comment (zburke) [reply src/RequestForm.js]: Looks good. I trust your judgement here.

**Pull request merged**